### PR TITLE
groups: dropdowns redesign on mobile

### PR DESCRIPTION
**Pull request by dantish**
Fixes LAND-713.

~**Important:** Right now it's implemented for the group actions only. Later, it will be extracted into a separate component (`Actions` probably) with a unified API to encapsulate the logic of showing different modals for web and mobile.~

### CODE REVIEW

**Comment by arthyn** [comment]
I have a branch that I started during standup yesterday with React 18 just about working so maybe we could just transition to importing this as an actual dep

**Comment by patosullivan** [comment]
> I have a branch that I started during standup yesterday with React 18 just about working so maybe we could just transition to importing this as an actual dep

that would be cool

**Comment by dantish** [comment]
Use `ActionsModal` for the edit section dropdown
```diff
diff --git a/ui/src/components/ActionsModal.tsx b/ui/src/components/ActionsModal.tsx
index 274e013b..ce68f998 100644
--- a/ui/src/components/ActionsModal.tsx
+++ b/ui/src/components/ActionsModal.tsx
@@ -109,7 +109,7 @@ const ActionsModal = React.memo(
             </DropdownMenu.Trigger>
             <DropdownMenu.Content
               align={align}
-              className={cn('dropdown min-w-52', contentClassName)}
+              className={cn('dropdown', contentClassName)}
             >
               {actions.map((action) => (
                 <DropdownMenu.Item
diff --git a/ui/src/components/Sidebar/__snapshots__/Sidebar.test.tsx.snap b/ui/src/components/Sidebar/__snapshots__/Sidebar.test.tsx.snap
index b912a662..0244893b 100644
--- a/ui/src/components/Sidebar/__snapshots__/Sidebar.test.tsx.snap
+++ b/ui/src/components/Sidebar/__snapshots__/Sidebar.test.tsx.snap
@@ -8,78 +8,80 @@ exports[`Sidebar > renders as expected 1`] = `
     <div
       class="flex w-full flex-col space-y-0.5 p-2"
     >
-      <div
-        class="group relative my-0.5 flex w-full items-center justify-between rounded-lg text-gray-800 sm:text-gray-600 hover:bg-gray-50"
-      >
-        <button
-          aria-expanded="false"
-          aria-haspopup="menu"
-          class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 appearance-none text-black hover:text-gray-800 group"
-          data-state="closed"
-          id="radix-0"
-          type="button"
+      <div>
+        <div
+          class="group relative my-0.5 flex w-full items-center justify-between rounded-lg text-gray-800 sm:text-gray-600 hover:bg-gray-50"
         >
-          <div
-            class="h-6 w-6 rounded group-hover:bg-gray-100"
+          <button
+            aria-expanded="false"
+            aria-haspopup="menu"
+            class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 appearance-none text-black hover:text-gray-800 group"
+            data-state="closed"
+            id="radix-0"
+            type="button"
           >
-            <svg
-              class="h-6 w-6 group-hover:hidden"
-              fill="none"
-              viewBox="0 0 24 24"
-              xmlns="http://www.w3.org/2000/svg"
-            >
-              <path
-                class="fill-current"
-                d="M11.0833 17.9583V16.5833C11.0833 16.0771 11.4937 15.6667 12 15.6667C12.5063 15.6667 12.9167 16.0771 12.9167 16.5833V17.9583C12.9167 20.7428 15.1739 23 17.9583 23C20.7428 23 23 20.7428 23 17.9583C23 15.1739 20.7428 12.9167 17.9583 12.9167H16.5833V13.8333C16.5833 14.3396 16.1729 14.75 15.6667 14.75C15.1604 14.75 14.75 14.3396 14.75 13.8333V10.1667C14.75 9.66041 15.1604 9.25 15.6667 9.25C16.1729 9.25 16.5833 9.66041 16.5833 10.1667V11.0833H17.9583C20.7428 11.0833 23 8.8261 23 6.04167C23 3.25723 20.7428 1 17.9583 1C15.1739 1 12.9167 3.25723 12.9167 6.04167V7.41667C12.9167 7.92293 12.5063 8.33333 12 8.33333C11.4937 8.33333 11.0833 7.92293 11.0833 7.41667V6.04167C11.0833 3.25723 8.8261 1 6.04167 1C3.25723 1 1 3.25723 1 6.04167C1 8.8261 3.25723 11.0833 6.04167 11.0833H7.41667V10.1667C7.41667 9.66041 7.82707 9.25 8.33333 9.25C8.83959 9.25 9.25 9.66041 9.25 10.1667V13.8333C9.25 14.3396 8.83959 14.75 8.33333 14.75C7.82707 14.75 7.41667 14.3396 7.41667 13.8333V12.9167H6.04167C3.25723 12.9167 1 15.1739 1 17.9583C1 20.7428 3.25723 23 6.04167 23C8.8261 23 11.0833 20.7428 11.0833 17.9583Z"
-              />
-            </svg>
-            <svg
-              class="m-1 h-4 w-4 text-gray-800 hidden group-hover:block"
-              fill="none"
-              viewBox="0 0 24 24"
-              xmlns="http://www.w3.org/2000/svg"
+            <div
+              class="h-6 w-6 rounded group-hover:bg-gray-100"
             >
-              <path
-                class="fill-current"
-                clip-rule="evenodd"
-                d="M5 6c0-.55228.44772-1 1-1h12c.5523 0 1 .44772 1 1s-.4477 1-1 1H6c-.55228 0-1-.44772-1-1Zm0 6c0-.5523.44772-1 1-1h12c.5523 0 1 .4477 1 1s-.4477 1-1 1H6c-.55228 0-1-.4477-1-1Zm1 5c-.55228 0-1 .4477-1 1s.44772 1 1 1h12c.5523 0 1-.4477 1-1s-.4477-1-1-1H6Z"
-                fill-rule="evenodd"
-              />
-            </svg>
-          </div>
-          <div
-            class="max-w-full flex-1 text-left text-lg font-bold sm:text-base sm:font-semibold  truncate text-gray-800 sm:text-gray-600"
-          >
+              <svg
+                class="h-6 w-6 group-hover:hidden"
+                fill="none"
+                viewBox="0 0 24 24"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  class="fill-current"
+                  d="M11.0833 17.9583V16.5833C11.0833 16.0771 11.4937 15.6667 12 15.6667C12.5063 15.6667 12.9167 16.0771 12.9167 16.5833V17.9583C12.9167 20.7428 15.1739 23 17.9583 23C20.7428 23 23 20.7428 23 17.9583C23 15.1739 20.7428 12.9167 17.9583 12.9167H16.5833V13.8333C16.5833 14.3396 16.1729 14.75 15.6667 14.75C15.1604 14.75 14.75 14.3396 14.75 13.8333V10.1667C14.75 9.66041 15.1604 9.25 15.6667 9.25C16.1729 9.25 16.5833 9.66041 16.5833 10.1667V11.0833H17.9583C20.7428 11.0833 23 8.8261 23 6.04167C23 3.25723 20.7428 1 17.9583 1C15.1739 1 12.9167 3.25723 12.9167 6.04167V7.41667C12.9167 7.92293 12.5063 8.33333 12 8.33333C11.4937 8.33333 11.0833 7.92293 11.0833 7.41667V6.04167C11.0833 3.25723 8.8261 1 6.04167 1C3.25723 1 1 3.25723 1 6.04167C1 8.8261 3.25723 11.0833 6.04167 11.0833H7.41667V10.1667C7.41667 9.66041 7.82707 9.25 8.33333 9.25C8.83959 9.25 9.25 9.66041 9.25 10.1667V13.8333C9.25 14.3396 8.83959 14.75 8.33333 14.75C7.82707 14.75 7.41667 14.3396 7.41667 13.8333V12.9167H6.04167C3.25723 12.9167 1 15.1739 1 17.9583C1 20.7428 3.25723 23 6.04167 23C8.8261 23 11.0833 20.7428 11.0833 17.9583Z"
+                />
+              </svg>
+              <svg
+                class="m-1 h-4 w-4 text-gray-800 hidden group-hover:block"
+                fill="none"
+                viewBox="0 0 24 24"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  class="fill-current"
+                  clip-rule="evenodd"
+                  d="M5 6c0-.55228.44772-1 1-1h12c.5523 0 1 .44772 1 1s-.4477 1-1 1H6c-.55228 0-1-.44772-1-1Zm0 6c0-.5523.44772-1 1-1h12c.5523 0 1 .4477 1 1s-.4477 1-1 1H6c-.55228 0-1-.4477-1-1Zm1 5c-.55228 0-1 .4477-1 1s.44772 1 1 1h12c.5523 0 1-.4477 1-1s-.4477-1-1-1H6Z"
+                  fill-rule="evenodd"
+                />
+              </svg>
+            </div>
             <div
-              class="flex items-center justify-between"
+              class="max-w-full flex-1 text-left text-lg font-bold sm:text-base sm:font-semibold  truncate text-gray-800 sm:text-gray-600"
             >
-              Groups
-              <a
-                aria-label="Back to Landscape"
-                class="h-6 w-6 no-underline hidden group-hover:block"
-                href="/apps/grid"
-                rel="noreferrer"
-                target="_blank"
-                title="Back to Landscape"
+              <div
+                class="flex items-center justify-between"
               >
-                <svg
-                  class="text-gray-400"
-                  fill="none"
-                  viewBox="0 0 24 24"
-                  xmlns="http://www.w3.org/2000/svg"
+                Groups
+                <a
+                  aria-label="Back to Landscape"
+                  class="h-6 w-6 no-underline hidden group-hover:block"
+                  href="/apps/grid"
+                  rel="noreferrer"
+                  target="_blank"
+                  title="Back to Landscape"
                 >
-                  <path
-                    class="stroke-current"
-                    d="M8 14V8M8 8H14M8 8L15 15"
-                    stroke-linecap="round"
-                    stroke-linejoin="round"
-                    stroke-width="2"
-                  />
-                </svg>
-              </a>
+                  <svg
+                    class="text-gray-400"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    xmlns="http://www.w3.org/2000/svg"
+                  >
+                    <path
+                      class="stroke-current"
+                      d="M8 14V8M8 8H14M8 8L15 15"
+                      stroke-linecap="round"
+                      stroke-linejoin="round"
+                      stroke-width="2"
+                    />
+                  </svg>
+                </a>
+              </div>
             </div>
-          </div>
-        </button>
+          </button>
+        </div>
       </div>
       <button
         class="flex w-full cursor-pointer flex-row space-x-2 px-1 text-gray-400 hover:text-gray-800"
@@ -352,29 +354,31 @@ exports[`Sidebar > renders as expected 1`] = `
                   <div
                     class="pr-1"
                   >
-                    <button
-                      aria-expanded="false"
-                      aria-haspopup="menu"
-                      aria-label="Groups Sort Options"
-                      class="default-focus flex h-6 w-6 items-center rounded text-base font-semibold hover:bg-gray-50 sm:p-1"
-                      data-state="closed"
-                      id="radix-1"
-                      type="button"
-                    >
-                      <svg
-                        class="h-6 w-6 text-gray-400 sm:h-4 sm:w-4"
-                        fill="none"
-                        viewBox="0 0 24 24"
-                        xmlns="http://www.w3.org/2000/svg"
+                    <div>
+                      <button
+                        aria-expanded="false"
+                        aria-haspopup="menu"
+                        aria-label="Groups Sort Options"
+                        class="appearance-none default-focus flex h-6 w-6 items-center rounded text-base font-semibold hover:bg-gray-50 sm:p-1"
+                        data-state="closed"
+                        id="radix-1"
+                        type="button"
                       >
-                        <path
-                          class="fill-current"
-                          clip-rule="evenodd"
-                          d="M17.207 4.793a1 1 0 0 0-1.414 0l-3 3a1 1 0 0 0 1.414 1.414L15.5 7.914V17.5a1 1 0 1 0 2 0V7.914l1.293 1.293a1 1 0 1 0 1.414-1.414l-3-3ZM8.5 5.5a1 1 0 0 0-2 0v9.586l-1.293-1.293a1 1 0 0 0-1.414 1.414l3 3a1 1 0 0 0 1.414 0l3-3a1 1 0 0 0-1.414-1.414L8.5 15.086V5.5Z"
-                          fill-rule="evenodd"
-                        />
-                      </svg>
-                    </button>
+                        <svg
+                          class="h-6 w-6 text-gray-400 sm:h-4 sm:w-4"
+                          fill="none"
+                          viewBox="0 0 24 24"
+                          xmlns="http://www.w3.org/2000/svg"
+                        >
+                          <path
+                            class="fill-current"
+                            clip-rule="evenodd"
+                            d="M17.207 4.793a1 1 0 0 0-1.414 0l-3 3a1 1 0 0 0 1.414 1.414L15.5 7.914V17.5a1 1 0 1 0 2 0V7.914l1.293 1.293a1 1 0 1 0 1.414-1.414l-3-3ZM8.5 5.5a1 1 0 0 0-2 0v9.586l-1.293-1.293a1 1 0 0 0-1.414 1.414l3 3a1 1 0 0 0 1.414 0l3-3a1 1 0 0 0-1.414-1.414L8.5 15.086V5.5Z"
+                            fill-rule="evenodd"
+                          />
+                        </svg>
+                      </button>
+                    </div>
                   </div>
                 </div>
               </div>
diff --git a/ui/src/groups/ChannelsList/EditSectionDropdown.tsx b/ui/src/groups/ChannelsList/EditSectionDropdown.tsx
index 14e2c974..c6cce375 100644
--- a/ui/src/groups/ChannelsList/EditSectionDropdown.tsx
+++ b/ui/src/groups/ChannelsList/EditSectionDropdown.tsx
@@ -1,7 +1,7 @@
 import React, { useState } from 'react';
-import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
 import ElipsisIcon from '@/components/icons/EllipsisIcon';
 import ConfirmationModal from '@/components/ConfirmationModal';
+import ActionsModal, { Action } from '@/components/ActionsModal';
 
 interface EditSectionDropDownProps {
   handleEditClick: () => void;
@@ -12,30 +12,31 @@ export default function EditSectionDropDown({
   handleEditClick,
   handleDeleteClick,
 }: EditSectionDropDownProps) {
+  const [open, setOpen] = useState(false);
   const [deleteOpen, setDeleteOpen] = useState(false);
+
+  const actions: Action[] = [
+    {
+      key: 'edit',
+      type: 'default',
+      onClick: handleEditClick,
+      content: 'Edit Section',
+    },
+    {
+      key: 'delete',
+      type: 'destructive',
+      onClick: () => setDeleteOpen(true),
+      content: 'Delete Section',
+    },
+  ];
+
   return (
     <>
-      <DropdownMenu.Root>
-        <DropdownMenu.Trigger asChild>
-          <div className="cursor-pointer">
-            <ElipsisIcon className="h-5 w-5 fill-gray-600" />
-          </div>
-        </DropdownMenu.Trigger>
-        <DropdownMenu.Content className="dropdown">
-          <DropdownMenu.Item
-            className="dropdown-item flex items-center space-x-2"
-            onSelect={handleEditClick}
-          >
-            <span>Edit Section</span>
-          </DropdownMenu.Item>
-          <DropdownMenu.Item
-            className="dropdown-item flex items-center space-x-2"
-            onSelect={() => setDeleteOpen(true)}
-          >
-            <span className="text-red">Delete Section</span>
-          </DropdownMenu.Item>
-        </DropdownMenu.Content>
-      </DropdownMenu.Root>
+      <ActionsModal open={open} onOpenChange={setOpen} actions={actions}>
+        <div className="cursor-pointer">
+          <ElipsisIcon className="h-5 w-5 fill-gray-600" />
+        </div>
+      </ActionsModal>
       <ConfirmationModal
         open={deleteOpen}
         setOpen={setDeleteOpen}

```

**Comment by jamesacklin** [comment]
Looks great, keep going!

**Comment by jamesacklin** [comment]
@dantish If you can, resolve ^ those merge conflicts ^ and we can merge to start playing with it on our phones. Thanks!

File: ui/src/components/ActionsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React, { PropsWithChildren, ReactNode } from 'react';
```
Comment (arthyn) [inline ui/src/components/ActionsModal.tsx:1]: `ActionMenu` might be a more clear name

**Review by arthyn — APPROVED** [review]
lgtm from a code perspective. would be nice if we could move to something closer to the API like radixui uses where we'd have something like:

```tsx
<ActionMenu>
  <ActionMenuTrigger />
  <ActionMenuContent>
    <ActionMenuIItem />
...
```

but not something that needs to block this

File: ui/src/dms/DMOptions.tsx [hunk 2]
```diff
@@ -100,86 +101,93 @@ export default function DmOptions({
...
+      content: pinned.includes(whom) ? 'Unpin' : 'Pin',
+    });
+
+    if (isMulti) {
+      actions.push({
+        key: 'invite',
+        onClick: handleInvite,
+        content: 'Invite',
+      });
+    }
```
[the remark this answers is not in the document]
Comment (dantish) [reply ui/src/dms/DMOptions.tsx:155]: They go in a specific order, so I can't just put them under a single if clause

[the remark this answers is not in the document]
Comment (arthyn) [reply ui/src/dms/DMOptions.tsx:155]: ah ok makes sense

**Comment by dantish** [comment]
> @dantish If you can, resolve ^ those merge conflicts ^ and we can merge to start playing with it on our phones. Thanks!

Will do it first thing tomorrow morning + will try to change the API to what @arthyn proposed.

**Comment by dantish** [comment]
@jamesacklin @arthyn please, take a look again, looks like that's it finally.
By "take a look",  I mean not only from the code perspective but from the user's one too, so we can be sure that all changes are okay and nothing has been broken 🙏

**Review by jamesacklin — APPROVED** [review]

**Pull request merged**